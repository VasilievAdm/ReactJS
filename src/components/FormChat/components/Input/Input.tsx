import React, { FC } from 'react';

interface InputProps {
  change: React.ChangeEventHandler<HTMLInputElement>;
  value: string;
}

export const Input: FC<InputProps> = (props) => {
  return (
    <input
      className="chat-list_input"
      type="text"
      value={props.value}
      onChange={props.change}
      placeholder="Enter chat name"
    />
  );
};
