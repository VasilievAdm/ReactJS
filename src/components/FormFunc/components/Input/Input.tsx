import React, { FC } from 'react';

interface InputProps {
  change: React.ChangeEventHandler<HTMLInputElement>;
  value: string;
}

export const Input: FC<InputProps> = (props) => {
  return (
    <input
      className="input"
      type="text"
      value={props.value}
      onChange={props.change}
      placeholder="Enter text"
    />
  );
};
