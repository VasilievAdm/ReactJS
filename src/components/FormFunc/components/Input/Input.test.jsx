import React from 'react';
import { Input } from './Input';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('Input', () => {
  it('render component', () => {
    render(<Input />);
  });

  it('render with snapshot', () => {
    const { asFragment } = render(<Input />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('render component with placeholder', () => {
    render(<Input />);
    expect(screen.getByPlaceholderText(/Enter text/)).toBeInTheDocument();
  });

  it('render component without error', () => {
    render(<Input />);
    expect(screen.queryByText('Error')).not.toBeInTheDocument();
  });

  it('render multiply components', () => {
    render(
      <>
        <Input />
        <Input />
      </>
    );

    expect(screen.queryAllByRole('textbox').length).toBe(2);
  });

  it('render component with text', () => {
    render(<Input value="some text" />);
    expect(screen.getByRole('textbox')).toHaveValue('some text');
  });
});
