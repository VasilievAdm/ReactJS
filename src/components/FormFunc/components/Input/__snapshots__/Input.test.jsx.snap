// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input render with snapshot 1`] = `
<DocumentFragment>
  <input
    class="input"
    placeholder="Enter text"
    type="text"
    value=""
  />
</DocumentFragment>
`;
